File: hw/quina/u3/hw29singleton/AlertSystem/AlarmService/src/main/java/ec/edu/espe/alarmservice/config/AlarmConfig.java
package ec.edu.espe.alarmservice.config;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.model.Updates;
import ec.edu.espe.alarmservice.utils.MongoConnection;
import org.bson.Document;
/**
 *
 * @author Maryuri Quiña, @ESPE
 */
public class AlarmConfig {

    private static AlarmConfig instance;
    private int minimumStock;
    private final MongoCollection<Document> configCollection;

    private AlarmConfig() {
        this.configCollection = MongoConnection.getDatabase().getCollection("SalesConfiguration");
        Document doc = configCollection.find().first();
        this.minimumStock = (doc != null) ? doc.getInteger("minimumStock") : 10;
    }

    public static AlarmConfig getInstance() {
        if (instance == null) {
            instance = new AlarmConfig();
        }
        return instance;
    }

    public int getMinimumStock() {
        return minimumStock;
    }

    public void updateMinimumStock(int newStock) {
        this.minimumStock = newStock;
        Document firstDoc = configCollection.find().first();
        if (firstDoc != null) {
            configCollection.updateOne(firstDoc, Updates.set("minimumStock", newStock));
        } else {
            configCollection.insertOne(new Document("minimumStock", newStock));
        }
    }
}

File: hw/quina/u3/hw29singleton/AlertSystemPy/model/alarm_config.py
from utils.mongo_connection import MongoConnection

class AlarmConfig:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(AlarmConfig, cls).__new__(cls)

            db = MongoConnection.get_database()
            cls._instance.collection = db["SalesConfiguration"]
            

            config = cls._instance.collection.find_one()
            cls._instance.minimum_stock = config["minimumStock"] if config else 10
        return cls._instance

    def update_minimum_stock(self, new_stock):
        self.minimum_stock = int(new_stock)
        self.collection.update_one({}, {"$set": {"minimumStock": self.minimum_stock}}, upsert=True)

    def get_minimum_stock(self):
        return self.minimum_stock